test/features/public_asset/data/network/public_asset_api_test.dart

```diff
@@ -0,0 +1,181 @@
+  final dioAdapter = DioAdapter(dio: dio);
+  final dioAdapterHeaders = {"accept": "application/json;jmapVersion=rfc-8621"};
+  final httpClient = HttpClient(dio);
+  final processingInvocation = ProcessingInvocation();
+  final requestBuilder = JmapRequestBuilder(httpClient, processingInvocation);
+  final identityId = IdentityId(Id('some-identity-id'));
+  final methodCallId = MethodCallId('c0');
+  final publicAsset = PublicAsset(
+    id: Id('abc123'),
+    blobId: Id('def456'),
+    size: 123,
+    contentType: 'image/jpeg',
+    publicURI: 'http://domain.com/public/abc123',
+    identityIds: {identityId: true}
+  );
+
+  final publicAssetApi = PublicAssetApi(httpClient, MockUuid());
+  final accountId = AccountFixtures.aliceAccountId;
+  final session = SessionFixtures.aliceSession;
+  final mapPublicAssetIdToUpdatingIdentityIds = {
+    publicAsset.id!: {identityId: null}
+  };
+  final updateObject = PatchObject({
+    '${PatchObject.identityIdsProperty}/${identityId.id.value}': null,
+  });
+  final method = SetPublicAssetMethod(accountId)
+    ..addUpdates({publicAsset.id!: updateObject});
+  
+  group('public asset api test:', () {
+    test(
+      'should complete without exception '
+      'when server returns updated public asset ids',
+    () async {
+      // arrange
+      final invocation = requestBuilder.invocation(
+        method,
+        methodCallId: methodCallId);
+      dioAdapter.onPost(
+        '',
+        (server) => server.reply(200, {
+          "sessionState": "abcdefghij",
+          "methodResponses": [[
+            method.methodName.value,
+            {
+              "accountId": accountId.id.value,
+              "newState": 'some-state',
+              "updated": {publicAsset.id?.value: null},
+            },
+            methodCallId.value
+          ]]
+        }),
+        data: {
+          "using": method.requiredCapabilities
+            .map((capability) => capability.value.toString())
+            .toList(),
+          "methodCalls": [
+            [
+              method.methodName.value,
+              {
+                "accountId": accountId.id.value,
+                "update": {publicAsset.id?.value: updateObject.toJson()},
+              },
+              methodCallId.value
+            ],
+          ]
+        },
+        headers: dioAdapterHeaders,
+      );
+      
+      // act
+      final response = (await (requestBuilder..usings(method.requiredCapabilities))
+        .build()
+        .execute())
+          .parse<SetPublicAssetResponse>(
+            invocation.methodCallId,
+            SetPublicAssetResponse.deserialize);
+      
+      // assert
+      expect(response?.updated, equals({publicAsset.id: null}));
+      await expectLater(
+        publicAssetApi.partialUpdatePublicAssets(
+          session,
+          accountId,
+          mapPublicAssetIdToUpdatingIdentityIds: mapPublicAssetIdToUpdatingIdentityIds),
+        completes);
```

Comment:
what is it?

Reply:
`completes` is to check if a `Future<void>` function completes without throwing any exception. See more [here](https://api.flutter.dev/flutter/package-matcher_expect/completes.html)